src/app/components/shared/datetime-formats/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { Spectator, createComponentFactory } from "@ngneat/spectator";
+import { SharedModule } from "@shared/shared.module";
+import { MockBawApiModule } from "@baw-api/baw-apiMock.module";
+import { DateTime, IANAZone, Settings, SystemZone } from "luxon";
+import { assertTooltip } from "@test/helpers/html";
+import { DatetimeComponent } from "./datetime.component";
+
+// I have created this interface for TypeScript LSP typing and auto completion
+// it should not be used outside of this file
+interface TestCase {
+  name: string;
+  value: Date | DateTime;
+  expectedText: string;
+  expectedTooltip: string;
+  expectedDateTimeAttribute: string;
+  date?: boolean;
+  time?: boolean;
+}
+
+// by setting the defaultZone to utc when we create DateTime objects from ISO strings
+// it doesn't implicitly localize the DateTime object to another timezone
+// this makes the object produced by the iso string predictable
+// eg. an iso string "2020-10-10T12:12:12Z" will always produce the object
+//     { year: 2020, month: 10, day: 10, hour: 12, minute: 12, second: 12 }
+// we re-mock the test runners timezone to a more complex one when the tests are run
+// however, during test setup it is easier to read the DateTime objects if we create
+// them as if the test runner is in utc+00:00
+const originalTestRunnerZone = Settings.defaultZone;
+Settings.defaultZone = "utc";
+
+// in the following test cases, all the dateTime objects are created in UTC+00:00
+// therefore, because we are mocking the test runners timezone to be Australia/Perth (UTC+08:00)
+// we should see that the date time text, tooltips and machine readable content is localized
+// to UTC+08:00. eg. We should see the hours increase by 8
+const testCases: TestCase[] = [
+  {
+    name: "Luxon DateTime outside test runners timezone",
+    // create a Luxon DateTime object in UTC+00:00
+    // we should see this date be localized to UTC+08:00 (the test runners timezone)
+    // we do not call setZone() here like the other tests because we want to create the DateTime object in UTC+00:00
+    // so that we can see it localized to UTC+08:00
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "2020-01-01 20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+  },
+  {
+    name: "Luxon DateTime in test runners timezone",
+    // by setting the timezone to Australia/Perth, we should see that the time value is not changed
+    // because the Luxon DateTime object is already in the same timezone as the test runner
+    // by using { keepLocalTime: true }, we ensure that the time is not changed when we change the timezone
+    // without using { keepLocalTime: true }, the DateTime object would be implicitly created as 20:10:11 +08:00
+    value: DateTime.fromISO("2020-01-01T12:10:11.321Z").setZone(
+      "Australia/Perth",
+      { keepLocalTime: true }
+    ),
```

Comment:
```suggestion
    value: DateTime.fromISO("2020-01-01T12:10:11.321+08:00").setZone("Australia/Perth"),
```

Setting up your data as it was is actually just incorrect. Sure you massage it into what you mean, but why write a UTC date, when you can just write a date in the correct offset? Any time I see `keepLocalTime` and you're not using it for massaging user input, then you're breaking the date.

Reply:
Further if this was really a test of a local time you should be constructing a local date from a local string!

```suggestion
    value: DateTime.fromISO("2020-01-01T12:10:11.321")
```